server/neptune/gateway/event/push_event_handler.go

```diff
@@ -149,18 +160,70 @@ func (p *PushHandler) generatePlanSteps(repoID string) []workflows.Step {
 	return workflowSteps
 }
 
-func (p *PushHandler) generateApplySteps(repoID string) []workflows.Step {
-	var workflowSteps []workflows.Step
-	projectConfig := p.GlobalCfg.DefaultProjCfg(p.Logger, repoID, "path", defaultWorkspace)
-	steps := projectConfig.Workflow.Apply.Steps
-	for _, step := range steps {
-		workflowSteps = append(workflowSteps, workflows.Step{
-			StepName:    step.StepName,
-			ExtraArgs:   step.ExtraArgs,
-			RunCommand:  step.RunCommand,
-			EnvVarName:  step.EnvVarName,
-			EnvVarValue: step.EnvVarValue,
-		})
+func (p *PushHandler) buildRoots(event Push, ctx context.Context) ([]*valid.MergedProjectCfg, error) {
```

Comment:
Seems like a lot to have all this logic in one file?  I think having a struct responsible for building this project config would be useful.